### PR TITLE
Fix accessible_by fires query for rules using association as condition.

### PR DESCRIPTION
**Pull request by albb0920**
see #552

In turns out this could be fixed by swapping the left hand side and right hand side of `==`.

### CODE REVIEW

**Comment by albb0920** [comment]
Rubocop doesn't like it, do you want `(@conditions.is_a?(Hash) && @conditions == {})` instead?

File: spec/cancan/model_adapters/active_record_adapter_spec.rb [hunk 1]
```diff
@@ -115,6 +115,17 @@ class User < ActiveRecord::Base
       expect(Article.accessible_by(@ability)).to eq([article])
     end
 
+    it 'does not fires query with accessible_by() for abilities defined with association' do
+      user = User.create!
+      @ability.can(:read, Article, user.articles) { true }
```
Comment (coorasse) [inline spec/cancan/model_adapters/active_record_adapter_spec.rb]: this is not how you write rules in cancancan. you should replace it with `can :read, Article, user_id: user.id`, as discussed also #552.

**Comment by albb0920** [comment]
I've updated the test case, does this look more reasonable?

**Comment by coorasse** [comment]
closed in #620

**Pull request closed without merging**